java/src/jmri/jmrit/symbolicprog/DecVariableValue.java

```diff
@@ -153,15 +153,17 @@ void updatedTextField() {
         } catch (java.lang.NumberFormatException ex) {
             newVal = 0;
         }
-        int transfer = Math.max(newVal - _offset, 0); // prevent negative values, especially in tests outside UI
+
+        newVal = newVal - _offset;
```

Comment:
No need to change, but you can use:
newVal -= _offset;

It makes it a little more clear that you want to update the variable itself and that it's not a mistake.